internal/addon/secrets.go

```diff
@@ -0,0 +1,41 @@
+package addon
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Target defines the name of an endpoint that will be available to store
+// signal data.
+type Target string
```

Comment:
Now that you are doing this change, I would prefer if we can change the name of this. I think (not sure) `Target` is a term used in logging to specify telemetry data destination. Is that right? Since for OpenTelemetry we will be using this for receivers, exporters and other components.. can we look for a more generic name or something?

Reply:
`Target` was the name we came up with back in the demo phase of MCOA, we picked it since it was agnostic to any technology. In Logging, we call them `Outputs`. Any suggestions? What do you think of `Endpoint`?

Reply:
`Endpoint` sounds much better thanks!